keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
 
-This workflow describes the automation supporting creating a new (greenfield) backend bucket. Accessing this bucket is covered in [Sharing COSI Created Buckets](#sharing-cosi-created-buckets). COSI determines that a request is for an existing bucket when `BucketRequest.bucketInstanceName` is specified.
+The COSI sidecar, which runs alongside each of the drivers, listens for Bucket events. All but the sidecar with the specified provisioner will ignore the Bucket object. Only the appropriate sidecar will make a gRPC request to its driver to create a Bucket in the OSP.
 
-Here is the workflow:
+More information about COSI gRPC API is [here](#cosi-grpc-api)
 
-+ COSI central controller sees a new `BucketRequest` (BR).
-+ COSI gets the `BR.BucketClass` (directly or via the matching default) and verifies that the BR's namespace is allowed.
-+ If BR.bucketInstanceName is empty, COSI creates the associated `Bucket`, filling in fields from the `BucketClass` and BR. Else, we have a brownfield BR.
-+ The sidecar sees the new `Bucket` and gRPC calls the driver to create a backend bucket.
-+ COSI updates the BR and `Bucket` instance to reference each other.
+```
+    COSI Driver (s3.amazonaws.com)
+    |------------------------------------------|
+    | grpc ProvisionerCreateBucket({           |
+    |      "name": "br-$uuid",                 |
+    |      "protocol": "s3",                   |
+    |      "parameters": {                     |
+    |         "key": "value"                   |
+    |      }                                   |
+    | })                                       |
+    |------------------------------------------|
+```
+Once the Bucket is successfully created in the OSP, sidecar sets `bucketAvailable=true` in the Bucket object. At this point, the Bucket is ready to be utilized by workloads.
 
-In addition, the cosi-node-adapter sees a new app pod references a BAR:
-+ cosi-node-adapter receives _NodeStageVolume_ request from kubelet.
-+ adapter add a finalizer to both the BAR.BA and the BAR.BR.
-+ the finalizer value is `pod.name`
+## Generating Access Credentials for Buckets
 
-## Sharing COSI Created Buckets
-This is the greenfield -> brownfield access use case, where COSI has created the `Bucket` instance and the driver has provisioned a new bucket. Now, we want to share access to this bucket in other namespaces.
+The following stakeholders are involved in the lifecycle of access credential generation:
 
-![ShareBucket Workflow](images/share-bucket.png)
+ - Users  - request access to buckets
+ - Admins - establish cluster wide access policies
 
-If the bucket sharing is all within the same namespace then each `BucketAccessRequest` (in the namespace) only needs to reference the existing BR.
+Access credentials are represented by BucketAccess objects. Here are the steps for creating a BucketAccess:
 
-Here is the workflow:
+  ![COSI Access Credential Creation](images/create-bucket-access.jpg)
```

Comment:
under the COSI Driver column it shows "s3.GrantAccess". Do you mean to name the protocol here?

Reply:
Yeah, good idea

Reply:
The diagram below, in User column, states "user creates BucketAccessRequest", but 1st step below says "User creates BucketClaim".